Add optimize-for-size flag to gulp command

This patch adds the --optimize-for-size flag to the gulp command in package.json
to improve build performance by optimizing for memory usage.

Index: code-server/lib/vscode/package.json
===================================================================
--- code-server.orig/lib/vscode/package.json
+++ code-server/lib/vscode/package.json
@@ -32,7 +32,7 @@
     "watch-extensionsd": "deemon npm run watch-extensions",
     "kill-watch-extensionsd": "deemon --kill npm run watch-extensions",
     "precommit": "node build/hygiene.js",
-    "gulp": "node --max-old-space-size=8192 ./node_modules/gulp/bin/gulp.js",
+    "gulp": "node --max-old-space-size=8192 --optimize-for-size ./node_modules/gulp/bin/gulp.js",
     "electron": "node build/lib/electron",
     "7z": "7z",
     "update-grammars": "node build/npm/update-all-grammars.mjs",